File: Study Plan/Algorithm I/Max_Arrea_Island.cpp
class Solution {
public:
    int dfs(vector<vector<int>>& grid,int i,int j)
    {
        if(i < 0 || i >= grid.size() || j < 0 || j >= grid[0].size())
            return 0;
        if(grid[i][j])
        {
            grid[i][j] = 0;
            int left = dfs(grid, i - 1, j);
            int right= dfs(grid, i + 1, j);
            int bot = dfs(grid, i, j - 1);
            int top= dfs(grid, i, j + 1);
            return 1 + left + top + bot + right; 
        }
        return 0;
    }
    int maxAreaOfIsland(vector<vector<int>>& grid) {
        int maxx = 0;
        for(int i = 0; i < grid.size(); i ++)
        {
            for(int j = 0;j < grid[i].size(); j ++)
            {
                if(grid[i][j])
                    maxx = max(maxx, dfs(grid, i, j));
            }
        }
        return maxx;
    }
};

File: Study Plan/Algorithm I/Max_Arrea_Island.py
class Solution:
    def dfs(self,grid,i,j):
        grid[i][j] = 0
        area = 1
        for row,col in [(i + 1, j), (i - 1, j) ,(i, j + 1), (i, j - 1)]:
            if row >= 0 and row < len(grid) and col>=0 and col < len(grid[row]) and grid[row][col] == 1:
                area+=self.dfs(grid, row, col) 
        return area
            
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        maxArea = 0
        for i in range(len(grid)):
            for j in range(len(grid[i])):
                if grid[i][j] == 1:
                    aArea = self.dfs(grid, i, j)
                    maxArea = max(maxArea, aArea)
        return maxArea
        